docs/design/persistent-storage.md

```diff
@@ -0,0 +1,220 @@
+
+`PersistenceRequirements` expresses the capabilities and performance attributes of the underlying storage.  Allow discovery
+of storage by its capabilities and use those capabilities as pod storage requirements.
+
+Kubernetes makes no guarantees at runtime that the underlying storage exists or is available.  High availability is left to the storage provider.
+
+## Goals
+
+* Allow administrators to describe available storage
+* Allow pod authors to discover and request persistence for their volumes
+* Enforce security through access control lists and securing storage to the same namespace as the pod volume
+* Enforce quotas through admission control
+* Ensure developers can rely on storage being available without being closely bound to a particular disk, server, network, or storage device.
+
+
+#### Allow admins to describe and create available storage
+
+Admins can describe storage with `PersistentStorageControllers` or create storage directly by using the `PersistentStorage` API.  Storage created manually still requires a PersistentStorageController and accompanying interface to allow management by the cluster.
+
+
+```
+	
+	PersistentStorage
+		TypeMeta
+		ObjectMeta
+			Name - internal name
+			Labels - labeled like Volume.PersistenceRequirements.Selector
+		Spec
+			VolumeSource
+				HostPath
+				EmptyDir
+				GitRepo	
+				GCEPersistentDisk { modes: ReadWriteOnce | ReadOnlyMany }
+				AWSElasticBlockStore { modes: ReadWriteOnce }
+				NFSMount { modes: ReadWriteOnce | ReadOnlyMany | ReadWriteMany }
+				*** many volume plugins
+			RetentionPolicy	 -- volumes can be orphaned. allow never delete.
+		Status
+			Phase
+			CurrentMounts	MountList
+			LastMounts		MountList
+			PodRef			ObjectReference -- the pod originally requesting persistence
+			
+
+	// Manages one type of PersistentStorage
+	PersistentStorageController
+		Name
+		Spec
+			VolumeSource
+				GCEPersistentDisk { modes: ReadWriteOnce | ReadOnlyMany, resources: 5gb 30iops, .48 throughput  }
+			MaxInstances  100
+			IncrementBy   10
+		Status
+			InstanceCount
+			
+
+	//
+	//  addition to VolumeSource
+	//	
+	type MountMode byte
+	
+	const (
+		ReadWriteOne MountMode = 1 << iota
+		ReadOnlyMany
+		ReadWriteMany
+	)
+	
+	func (vs *VolumeSource) canMount(mode MountMode) bool {
+		return vs.modes & mode != 0
+	}
+	
+```
+
+> To manually make storage available, the admin would, for example, create a GCE VM, create, attach, and format 20 persistent disks.  The IDs are saved and the admin then creates 20 persistent-disk.json files and POSTs each GCE_PD to the API.  There are many ways to script adding storage to a cluster.
+
+
+
+#### Pod to specify storage requirements
+
+```
+
+	Pod.Spec.Volumes.Volume 
+			Name
+			VolumeSource -- to be bound with available storage from the pool
+			PersistenceRequirements
+				Mode				ReadWriteOnce | ReadOnlyMany
+				RetentionPolicy		DeleteOnPodExit | RetainOnPodExit
```

Comment:
@a-robinson I'm thinking about keeping the field name "RetentionPolicy" on both API structs because the term is appropriate, but two internal types for constant values:  StoreRetentionPolicy for admins to cull orphaned resources and VolumeRetentionPolicy for use by pod authors to keep or recycle their persistent volume.